### DISCUSSION TITLE
"Upload buffer miscompare"

### DISCUSSION THREAD

**Discussion by SmartKobol** [arendst/Tasmota#12586] [Q&A] [answered]
Dear Friends, When I try to upgrade Sonoff Basic, even R3, devices from 9.4.0 to 9.5.0 I got the message "upload buffer miscompare" . Practically all versions are the flashed ones. When flashed the switches "Belkin WeMo" ticked. Nevertheless TasmoAdmin allows upgrade and gives successful report, but the version stays the old one. Is it some HA / OTA configuration problem, firewall block? Please help to solve the problem.
![error message](https://user-images.githubusercontent.com/81106324/124772386-4692f500-df3c-11eb-9f47-02387b2c011b.JPG)

**Comment by sfromis** [comment, marked as answer]
Always use the compressed .gz binaries for OTA upgrading. Preferably pointing directly at the URL on the OTA site, instead of downloading first.

**Reply by ascillato2** [reply]
Please, use minimal as explained in the docs.

Please, read the summary https://github.com/arendst/Tasmota/discussions/12579#discussioncomment-970413

**Reply by ascillato2** [reply]
Copy and Paste:

## Summary:

For upgrading, please, do as following:

### Please, Don't flash minimal as the first firmware.

### Please, don't update from an old version using minimal to minimal.

### Please, follow the upgrade steps https://tasmota.github.io/docs/Upgrading/

### If you see the message **_"upload buffer miscompare"_** means not enough space. Please, use minimal or lite first.

**_(Not following these steps can end up with a non-working device and you will need to erase all flash and flash by serial)_**

### UPGRADE OTAurl WAY:

**Please, don't use minimal in OTAurl**. Tasmota will take care of downloading and upgrading the minimal version automatically in-between updates. Just point to the Full version you want.

+ Upgrade from OLD Full Version -> New Full Version
   + Example: From Full 8.5.1 -> Full 9.1 -> 9.5 FULL
          + Set the following OTAurl in Tasmota UI and click on OTAUrl Upgrade
                http://ota.tasmota.com/tasmota/release-9.1.0/tasmota.bin.gz
          + (WAIT 2 to 5 minutes and don't reset the device)
          + Set the following OTAurl in Tasmota UI and click on OTAUrl Upgrade
                http://ota.tasmota.com/tasmota/tasmota.bin.gz
          + (WAIT 2 to 5 minutes and don't reset the device)

### UPGRADE MANUAL WAY:

**Please, don't update from an old version using minimal to minimal.**
+ Upgrade from OLD Full Version -> New minimal version -> New Full Version
   + Example: Full 8.5.1 -> minimal 9.1 -> Full 9.1 -> 9.5 minimal -> 9.5 FULL
+ Alternatively: Full 8.51 -> minimal 9.1 -> Lite 9.1 -> Lite 9.5

_(For more information on this topic, please refer to the Tasmota Docs at https://tasmota.github.io/docs/Upgrading/ and older issues. Thanks.)_

**Reply by SmartKobol** [reply]
Dear ascillato2, Thank you for the detailed answer. I will follow the points. 

> 
> 
> Always use the compressed .gz binaries for OTA upgrading. Preferably pointing directly at the URL on the OTA site, instead of downloading first.

**Reply by SmartKobol** [reply]
Dear sfromis, Thank you for the orientation. I do appreciate!
